New PIL version broke tests

After merging a green PR all unit tests on `branch-3.0` are failing:
```python
______________________ Test_Image.test_transform_PIL[gif] ______________________
self = <test_visual.Test_Image object at 0x7f7dadf9e310>, typ = 'gif'
@pytest.mark.parametrize('typ', ('png', 'gif', 'tiff'))
deftest_transform_PIL(self, typ) -> None:
        image = PIL.Image.new("RGBA", size=(50, 50), color=(155, 0, 0))
        out = BytesIO()
        image.save(out, typ)
        value = PIL.Image.open(out)
        expected = "data:image/%s;base64," % typ + base64.b64encode(out.getvalue()).decode('ascii')
        prop = bcpv.Image()
>       assert prop.transform(value) == expected
E       AssertionError: assert 'data:image/g...KtKjRhwEBADs=' == 'data:image/g...0qNGHAQEAOw=='
E         Skipping 46 identical leading characters in diff, use -v to show
E         - AAAAAAAAACH5BAEAAAEALAAAAAAyADIAQAhTAAEIHEiwoMGDCBMqXMiwocOHECNKnEixosWLGDNq3Mixo8ePIEOKHEmypMmTKFOqXMmypcuXMGPKnEmzps2bOHPq3Mmzp8+fQIMKHUq0qNGHAQEAOw==
E         + AAAAAAAAACwAAAAAMgAyAEAIUwABCBxIsKDBgwgTKlzIsKHDhxAjSpxIsaLFixgzatzIsaPHjyBDihxJsqTJkyhTqlzJsqXLlzBjypxJs6bNmzhz6tzJs6fPn0CDCh1KtKjRhwEBADs=
tests/unit/bokeh/core/property/test_visual.py:295: AssertionError
```
The last passing test had `pillow==9.1.1` whereas the failing tests have `pillow==9.2.0`

Not sure if this is purely a test issue to update or of there is a real change in `pillow` that we will need to accommodate in the library code.

--- a/tests/unit/bokeh/core/property/test_visual.py
+++ b/tests/unit/bokeh/core/property/test_visual.py
@@ -21,6 +21,7 @@ import base64
 import datetime
 from io import BytesIO
 from pathlib import Path
+from typing import Literal
 
 # External imports
 import numpy as np
@@ -284,12 +285,15 @@ class Test_Image:
         assert prop.transform(data) == expected
 
     @pytest.mark.parametrize('typ', ('png', 'gif', 'tiff'))
-    def test_transform_PIL(self, typ) -> None:
+    def test_transform_PIL(self, typ: Literal["png", "gif", "tiff"]) -> None:
         image = PIL.Image.new("RGBA", size=(50, 50), color=(155, 0, 0))
-        out = BytesIO()
-        image.save(out, typ)
-        value = PIL.Image.open(out)
-        expected = "data:image/%s;base64," % typ + base64.b64encode(out.getvalue()).decode('ascii')
+        out0 = BytesIO()
+        image.save(out0, typ)
+
+        value = PIL.Image.open(out0)
+        out1 = BytesIO()
+        value.save(out1, typ)
+        expected = "data:image/%s;base64," % typ + base64.b64encode(out1.getvalue()).decode('ascii')
 
         prop = bcpv.Image()
         assert prop.transform(value) == expected

